test/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1461 @@
+  const REVERT_MESSAGE_IF_PAYMENT_REVOCATION_COUNTER_HAS_REACHED_MAXIMUM =
+    "CardPaymentProcessor: revocation counter of the payment has reached the configured maximum";
+
+  let cardPaymentProcessor: Contract;
+  let tokenMock: Contract;
+  let deployer: SignerWithAddress;
+  let user1: SignerWithAddress;
+  let user2: SignerWithAddress;
+  let ownerRole: string;
+  let pauserRole: string;
+  let executorRole: string;
+
+  async function setUpContractsForPayments(payments: TestPayment[]) {
+    for (let payment of payments) {
+      await proveTx(tokenMock.mint(payment.account.address, payment.amount));
+      const allowance: BigNumber = await tokenMock.allowance(payment.account.address, cardPaymentProcessor.address);
+      if (allowance.lt(BigNumber.from(ethers.constants.MaxUint256))) {
+        await proveTx(tokenMock.connect(payment.account).approve(
+          cardPaymentProcessor.address,
+          ethers.constants.MaxUint256
+        ));
+      }
+    }
+  }
+
+  async function makePayments(payments: TestPayment[]) {
+    for (let payment of payments) {
+      await proveTx(cardPaymentProcessor.connect(payment.account).makePayment(
+        payment.amount,
+        createBytesString(payment.authorizationId, BYTES16_LENGTH),
+        createBytesString(payment.makingPaymentCorrelationId, BYTES16_LENGTH)
+      ));
+      payment.status = PaymentStatus.Uncleared;
+    }
+  }
+
+  async function setExecutorRole(account: SignerWithAddress) {
+    await proveTx(cardPaymentProcessor.grantRole(executorRole, account.address));
+  }
+
+  async function clearPayments(payments: TestPayment[], adminAccount: SignerWithAddress) {
+    const authorizationIds: string[] = [];
+    payments.forEach((payment: TestPayment) => {
+      authorizationIds.push(createBytesString(payment.authorizationId, BYTES16_LENGTH));
+      payment.status = PaymentStatus.Cleared;
+    });
+    await proveTx(cardPaymentProcessor.connect(adminAccount).clearPayments(authorizationIds));
+  }
+
+  function defineBalancesPerAccount(payments: TestPayment[], targetPaymentStatus: PaymentStatus): Map<string, number> {
+    const balancesPerAccount: Map<string, number> = new Map<string, number>();
+
+    payments.forEach((payment: TestPayment) => {
+      const address: string = payment.account.address;
+      let newBalance: number = balancesPerAccount.get(address) || 0;
+      if (payment.status == targetPaymentStatus) {
+        newBalance += payment.amount;
+      }
+      balancesPerAccount.set(address, newBalance);
+    });
+
+    return balancesPerAccount;
+  }
+
+  function defineExpectedCardPaymentProcessorState(payments: TestPayment[]): CardPaymentProcessorState {
+    const tokenBalance: number = countNumberArrayTotal(
+      payments.map(
+        function (payment: TestPayment): number {
+          return payment.status == PaymentStatus.Uncleared || payment.status == PaymentStatus.Cleared
```

Comment:
Codacy has a fix for the issue: [Replace `··········return·payment.status·==·PaymentStatus.Uncleared·||` with `········return·payment.status·==·PaymentStatus.Uncleared·||⏎·········`](https://app.codacy.com/gh/cloudwalk/brlc-periphery/pullRequest?prid=9942053)
```suggestion
        return payment.status == PaymentStatus.Uncleared ||
```